### DISCUSSION TITLE
npm run watch doesn't pickup classes defined as PHP variables

### DISCUSSION THREAD

**Discussion by MarketHubb** [tailwindlabs/tailwindcss#11298] [Help]
_This is probably easy, but I'm new to this development style. I really appreciate your help_

I have some badge and other element colors generated from user-selected CMS values. Here's a portion of the template that outputs custom badges:

`$badges .= '<span class="inline-flex items-center rounded-md bg-' . $badge_color . ' ';`

The (span) element and all classes are outputting correctly on the page, but `npm run watch` doesn't appear to be picking the variable-defined class names like `bg-blue-400`, so they aren't styled correctly. 

If I hardcode the _exact_ same badge into the template, npm does pick it up.
 
What do you think is the best way around this?

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

You could:
 - Have the entire class in `$badge_color` like
   ```php
   $badge_color = 'bg-blue-400';
   ```
   Ensure this code appears in a file covered by the `content` file globs.
 - Use the `style` attribute for truly dynamic colors, if `$badge_color` can be converted to a valid CSS color value:
   ```php
   $badges .= '<span class="…" style="background-color:' . $badge_color . '">';
   ```
 - [`safelist` the classes](https://tailwindcss.com/docs/content-configuration#safelisting-classes), if you have a limited number of known colors:
   ```js
   module.exports = {
     safelist: [
       'bg-blue-400',
       // …
     ],
     // …
   ];
   ```
